hls-test-utils/src/Development/IDE/Test/Diagnostic.hs

```diff
@@ -17,9 +18,9 @@ type ErrorMsg = String
 requireDiagnostic
     :: (Foldable f, Show (f Diagnostic), HasCallStack)
     => f Diagnostic
-    -> (DiagnosticSeverity, Cursor, T.Text, Maybe DiagnosticTag)
+    -> (DiagnosticSeverity, Cursor, T.Text, Maybe T.Text, Maybe DiagnosticTag)
```

Comment:
I think `Maybe T.Text` is not a good assertion type. It works for now, but it will get annoying once new error types are added, then we need awkward conditionals in the test cases.

In the future, we might want to have something like 

```haskell
data ErrorCode = NoCode | GhcCode Int | GhcCodeSince GhcVersion Int
```

However, this is not a blocker for this PR, I mostly want to know your opinion on this.

Reply:
It seems like a good idea. But I think we should indeed defer that to a future PR, because it requires parsing the error code, and I think we might not only want GHC codes in the future, for example cabal also has diagnostic codes.

Reply:
I want to avoid the parsing by using different constructors for each error code type, e.g. `GhcCode` vs `CabalCode` and so on.

Reply:
I mean the LSP diagnostics we get are strings/text, so we'd need to parse that.